janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/JanusGraphIndexTest.java

```diff
@@ -3080,9 +3084,18 @@ public void shouldUpdateIndexFieldsAfterIndexModification() throws InterruptedEx
         mgmt.updateIndex(mgmt.getGraphIndex(indexName), SchemaAction.REINDEX).get();
         mgmt.commit();
 
-        vertex = graph.addVertex(T.label, vertexL, key1, 1L, key2, 2L, key3, 3L);
+        graph2.addVertex(T.label, vertexL, key1, 1L, key2, 2L, key3, 3L);
+        JanusGraphException ex = assertThrows(JanusGraphException.class, () -> graph2.tx().commit());
```

Comment:
This exception only occurs because `graph2` did not have enough time to adopt the schema change yet, right? If so, can we be sure that this has not happened yet?
I think although it is a good idea to have a test for this behavior, this one could easily fail randomly, if the schema change is adopted early.

Reply:
You are right, but I cannot come up with a non-intrusive way to make sure `graph2` hasn't be able to read from `ManagementLogger` and adopt the schema changes. FWIW, the interval for ManagementLogger::read call is defined in GraphDatabaseConfiguration::LOG_READ_INTERVAL, which is by default 5 seconds. That's also why we need to wait for 5 seconds below.

I ran this test in GitHub CI for 3 times and many times locally. So far, so good. That being said, let me know if you have a better way to improve the resilience of this test.

Reply:
The only way I can imagine is by introducing some very ugly `synchronized` blocks and locks within the `ManagementLogger::read` method. But what about marking this test as flaky?

Reply:
I marked this test as `RepeatedIfExceptionsTest`.

Reply:
That would cover the case where 5 seconds are not enough time for the synchronization process. My concerns were about the first test. But as I have no idea how to set up a test environment for the first case either, I am fine with leaving it as is.

Reply:
`RepeatedIfExceptionsTest` is intended to cover the first test assertion. If L3088 does not throw an exception, then JUnit will throw an `AssertionFailedError`, which leads to a retry of the whole test. The scenario for L3088 to fail the test is that schema changes might get adopted too soon. Note that schema changes are read every 5 seconds. Thus, the chance for the first case to fail is existent but small. By retrying 5 times, we reduce the possibility to a very minimal level.

Reply:
Alright then. I did not know that `@RepeatedIfExceptionsText` also repeats the test for assertion errors.